tests/rptest/tests/datalake/catalog_service_factory.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.catalog_service import CatalogType, CatalogService, CatalogImpl
+from rptest.services.nessie_catalog import NessieCatalog
+from rptest.services.apache_iceberg_catalog import IcebergRESTCatalog
+
+from typing import List, Tuple, Optional
+
+SUPPORTED_CATALOGS = [NessieCatalog, IcebergRESTCatalog]
+SUPPORTED_CATALOG_TYPES_AND_IMPLS = [(CatalogType.REST, CatalogImpl.JDBC),
+                                     (CatalogType.REST, CatalogImpl.HADOOP)]
```

Comment:
nit: mind calling out in the commit message that Nessie isn't here intentionally (if it is)